packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -525,17 +525,21 @@ const arrayNodePrototype = Object.create(Object.prototype, arrayNodePrototypePro
 // #endregion
 
 /**
- * Helper to coerce property keys to integer indexes (or undefined if not an in-range integer).
+ * Attempts to coerce the given property key to an integer index property.
+ * @param key - The property key to coerce.
+ * @param exclusiveMax - This restricts the range in which the resulting index is allowed to be.
+ * The coerced index of `key` must be less than `exclusiveMax` or else this function will return `undefined`.
+ * This is useful for reading an array within the bounds of its length, e.g. `asIndex(key, array.length)`.
```

Comment:
This code will convert things like "0x5" or "5.0" or "   5   " all to 5. It will convert empty strings and whitespace to 0 as well.

Theoretically we could have a customized array with a session local member named "" (its a valid field name, we even use it in some places, like as the underlying field on an array node), and this will incorrectly think thats trying to access the "0" field.

To be robust, the easiest thing is coercing the number back to a string and see if you get the exact same string as input.

Anyway, given how this function behaves, I don't think your new docs are really accurate since this isn't the right correct behavior for array indexing.

Reply:
Haha, yikes, I'll add some tests for that.